frontend/relay-workflows-lib/lib/components/SingleWorkflowView.tsx

```diff
@@ -1,170 +1,18 @@
-  };
-
-  const onArtifactHover = useCallback(
-    (artifact: Artifact | null) => {
-      setFilledTaskName(artifact ? artifact.parentTask : null);
-    },
-    [setFilledTaskName],
-  );
-
-  useEffect(() => {
-    setSelectedTasks(tasknames ? tasknames : []);
-  }, [tasknames, setSelectedTasks]);
-
-  useEffect(() => {
-    const filteredTasks = selectedTasks.length
-      ? selectedTasks
-          .map((name) => fetchedTasks.find((task) => task.name === name))
-          .filter((task): task is Task => !!task)
-      : fetchedTasks;
-    setArtifactList(filteredTasks.flatMap((task) => task.artifacts));
-  }, [selectedTasks, fetchedTasks]);
-
-  useEffect(() => {
-    const newOutputTasks: string[] = [];
-    const traverse = (tasks: TaskNode[]) => {
-      const sortedTasks = [...tasks].sort((a, b) =>
-        a.name.localeCompare(b.name),
-      );
-      sortedTasks.forEach((taskNode) => {
-        if (
-          taskNode.children &&
-          taskNode.children.length == 0 &&
-          !newOutputTasks.includes(taskNode.name)
-        ) {
-          newOutputTasks.push(taskNode.name);
-        } else if (taskNode.children && taskNode.children.length > 0) {
-          traverse(taskNode.children);
-        }
-      });
-    };
-    traverse(taskTree);
-    setOutputTasks(newOutputTasks);
-  }, [taskTree]);
-
-  return (
-    <>
-      <Box
-        sx={{
-          position: "relative",
-          display: "inline-flex",
-          alignItems: "flex-start",
-          width: "100%",
-          height: "100%",
-        }}
-      >
-        <Box
-          sx={{
-            display: "flex",
-            flexDirection: "row",
-            width: "100%",
-            height: "100%",
-            gap: 2,
-          }}
-        >
-          <Box
-            display="flex"
-            flexDirection="column"
-            gap={1}
-            sx={{ position: "absolute", left: "-100px" }}
-          >
-            <ToggleButton
-              value="output"
-              aria-label="output"
-              onClick={handleSelectOutput}
-            >
-              OUTPUT
-            </ToggleButton>
-            <ToggleButton
-              value="output"
-              aria-label="output"
-              onClick={handleSelectClear}
-            >
-              CLEAR
-            </ToggleButton>
-          </Box>
-          {workflowData && (
-            <WorkflowRelay
-              workflowName={workflowData.workflow.name}
-              visit={workflowData.workflow.visit}
-              workflowLink
-              filledTaskName={filledTaskName}
-              expanded={true}
-            />
-          )}
-        </Box>
-      </Box>
-      {tasknames && (
-        <TaskInfo
-          artifactList={artifactList}
-          onArtifactHover={onArtifactHover}
-        />
-      )}
-      {workflowData && <WorkflowInfo workflow={workflowData.workflow} />}
-    </>
+export default function SingleWorkflowView(props: SingleWorkflowViewProps) {
```

Comment:
does typescript have a mechanism to swap in the mock code at test time so you don't have to modify the production component like this? that would be nicer if it is easy to achieve.